TestSsDvMultiRangeQuery.testDuelWithStandardDisjunction fails

### Description

```
TestSsDvMultiRangeQuery > testDuelWithStandardDisjunction FAILED
    java.lang.AssertionError: expected:<1002+ hits> but was:<1001+ hits>
        at __randomizedtesting.SeedInfo.seed([5ECF78EE8ABF1FBA:DCCCA12D922C2A7E]:0)
        at junit@4.13.1/org.junit.Assert.fail(Assert.java:89)
        at junit@4.13.1/org.junit.Assert.failNotEquals(Assert.java:835)
        at junit@4.13.1/org.junit.Assert.assertEquals(Assert.java:120)
        at junit@4.13.1/org.junit.Assert.assertEquals(Assert.java:146)
        at org.apache.lucene.sandbox.search.TestSsDvMultiRangeQuery.testDuelWithStandardDisjunction(TestSsDvMultiRangeQuery.java:130)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at randomizedtesting.runner@2.8.1/com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
```

This then causes some threads to leak and the class runner to fail:

```
TestSsDvMultiRangeQuery > classMethod FAILED
    com.carrotsearch.randomizedtesting.ThreadLeakError: 1 thread leaked from SUITE scope at org.apache.lucene.sandbox.search.TestSsDvMultiRangeQuery:
       1) Thread[id=34, name=pool-1-thread-1, state=TIMED_WAITING, group=TGRP-TestSsDvMultiRangeQuery]
            at java.base/jdk.internal.misc.Unsafe.park(Native Method)
            at java.base/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:410)
            at java.base/java.util.concurrent.LinkedTransferQueue$DualNode.await(LinkedTransferQueue.java:452)
            at java.base/java.util.concurrent.SynchronousQueue$Transferer.xferLifo(SynchronousQueue.java:194)
            at java.base/java.util.concurrent.SynchronousQueue.xfer(SynchronousQueue.java:233)
            at java.base/java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:336)
            at java.base/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1069)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
            at java.base/java.lang.Thread.run(Thread.java:1570)
        at __randomizedtesting.SeedInfo.seed([5ECF78EE8ABF1FBA]:0)
```

### Gradle command to reproduce

```
./gradlew test --tests TestSsDvMultiRangeQuery.testDuelWithStandardDisjunction -Dtests.seed=5ECF78EE8ABF1FBA -Dtests.nightly=true -Dtests.locale=cv-RU -Dtests.timezone=US/Aleutian -Dtests.asserts=true -Dtests.file.encoding=UTF-8
```

Some additional seeds:

> java.lang.AssertionError: expected:<1003+ hits> but was:<1001+ hits>
```
./gradlew :lucene:sandbox:test --tests "org.apache.lucene.sandbox.search.TestSsDvMultiRangeQuery.testDuelWithStandardDisjunction" -Ptests.jvms=12 -Ptests.jvmargs= -Ptests.seed=99B3647F731B035E -Ptests.useSecurityManager=true -Ptests.nightly=true -Ptests.gui=true -Ptests.file.encoding=US-ASCII -Ptests.vectorsize=512 -Ptests.forceintegervectors=true
```

> java.lang.AssertionError: expected:<1002+ hits> but was:<1001+ hits>
```
gradlew :lucene:sandbox:test --tests "org.apache.lucene.sandbox.search.TestSsDvMultiRangeQuery.testDuelWithStandardDisjunction" -Ptests.jvms=12 -Ptests.jvmargs= -Ptests.seed=671F5F892B420491 -Ptests.useSecurityManager=true -Ptests.nightly=true -Ptests.gui=true -Ptests.file.encoding=UTF-8 -Ptests.vectorsize=512 -Ptests.forceintegervectors=true
```

--- a/lucene/sandbox/src/test/org/apache/lucene/sandbox/search/TestSsDvMultiRangeQuery.java
+++ b/lucene/sandbox/src/test/org/apache/lucene/sandbox/search/TestSsDvMultiRangeQuery.java
@@ -72,7 +72,8 @@ public class TestSsDvMultiRangeQuery extends LuceneTestCase {
       }
 
       long[] scratch = new long[dims];
-      for (int i = 0; i < (LuceneTestCase.TEST_NIGHTLY ? atLeast(1000) : 100); i++) {
+      int maxDocs = LuceneTestCase.TEST_NIGHTLY ? atLeast(1000) : 100;
+      for (int i = 0; i < maxDocs; i++) {
         int numPoints = singleton ? 1 : RandomNumbers.randomIntBetween(random(), 1, 10);
         Document doc = new Document();
         for (int j = 0; j < numPoints; j++) {
@@ -124,9 +125,9 @@ public class TestSsDvMultiRangeQuery extends LuceneTestCase {
       Query query1 = builder1.build();
       Query query2 = builder2.build();
       Query query3 = builder3.build();
-      TopDocs result1 = searcher.search(query1, 100, Sort.INDEXORDER);
-      TopDocs result2 = searcher.search(query2, 100, Sort.INDEXORDER);
-      TopDocs result3 = searcher.search(query3, 100, Sort.INDEXORDER);
+      TopDocs result1 = searcher.search(query1, maxDocs, Sort.INDEXORDER);
+      TopDocs result2 = searcher.search(query2, maxDocs, Sort.INDEXORDER);
+      TopDocs result3 = searcher.search(query3, maxDocs, Sort.INDEXORDER);
       assertEquals(result2.totalHits, result1.totalHits);
       assertEquals(result2.totalHits, result3.totalHits);
       assertEquals(result2.scoreDocs.length, result1.scoreDocs.length);

